integration-tests/src/test/java/gov/uk/address/api/stepDefinitions/AddressApiHappyPath.java

```diff
@@ -322,6 +324,29 @@ public void aValidJWTIsReturnedInTheResponse() throws ParseException, IOExceptio
         makeAssertions(SignedJWT.parse(response.body()));
     }
 
+    @When("use sends a GET request to Addresses end point")
+    public void userSendsAGetRequestToAddressesEndpoint()
+            throws URISyntaxException, IOException, InterruptedException {
+        var request =
+                HttpRequest.newBuilder()
+                        .uri(new URIBuilder(getPrivateAPIEndpoint()).setPath(GET_ADDRESSES).build())
+                        .setHeader("session_id", sessionId)
+                        .GET()
+                        .build();
+
+        this.response = sendHttpRequest(request);
+    }
+
+    @Then("user should receive a valid response")
+    public void userShouldReceiveAValidResponse() throws ParseException, IOException {
+        assertEquals(200, response.statusCode());
+        var payload = objectMapper.readTree(response.body().toString());
+
+        assertEquals(
+                "MK42 7AA",
```

Comment:
Not sure where this is being set from. I logged out the response and it had a postcode set, but a global search of where this postcode came from resulted in nothing...

Reply:
The postcode is actually fetched from the user identity which we get in the initial steps of the test [here](https://github.com/alphagov/di-ipv-cri-address-api/blob/19489cb916410b2822e25c90a4181871ed86e5ab/integration-tests/src/test/java/gov/uk/address/api/stepDefinitions/AddressApiHappyPath.java#L71)

You can pick the json object from the above step to find the postcode and assert that here.
E.g.
`
    Map<String, String> deserializedResponse =
                objectMapper.readValue(userIdentityJson, new TypeReference<>() {});
        String inputPostCode = deserializedResponse.get("addresses").get("postcode");
`

and in test:
 assertEquals(
                inputPostCode, payload.get("result").get("addresses").get(0).get("postalCode").asText());